tests/scattering1d/test_keras_scattering1d.py

```diff
@@ -5,38 +5,45 @@
 import os
 import numpy as np
 import io
+import sys
 
 def test_Scattering1D():
-  """
-  Applies scattering on a stored signal to make sure its output agrees with
-  a previously calculated version.
-  """
-  test_data_dir = os.path.dirname(__file__)
-  with open(os.path.join(test_data_dir, 'test_data_1d.npz'), 'rb') as f:
-      buffer = io.BytesIO(f.read())
-      data = np.load(buffer)
-  x = data['x']
-  J = data['J']
-  Q = data['Q']
-  Sx0 = data['Sx']
-  # default
-  inputs0 = Input(shape=(x.shape[-1]))
-  scat0 = Scattering1D(J=J, Q=Q)(inputs0)
-  model0 = Model(inputs0, scat0)
-  model0.compile(optimizer='adam',
-                loss='sparse_categorical_crossentropy',
-                metrics=['accuracy'])
-  Sg0 = model0.predict(x)
-  assert np.allclose(Sg0, Sx0, atol=1e-07)
-  # adjust T
-  sigma_low_scale_factor = 2
-  T=2**(J-sigma_low_scale_factor)
-  inputs1 = Input(shape=(x.shape[-1]))
-  scat1 = Scattering1D(J=J, Q=Q, T=T)(inputs1)
-  model1 = Model(inputs1, scat1)
-  model1.compile(optimizer='adam',
-                loss='sparse_categorical_crossentropy',
-                metrics=['accuracy'])
-  Sg1 = model1.predict(x)
-  assert Sg1.shape == (Sg0.shape[0], Sg0.shape[1], Sg0.shape[2]*2**(sigma_low_scale_factor))
-  
\ No newline at end of file
+    """
+    Applies scattering on a stored signal to make sure its output agrees with
+    a previously calculated version.
+    """
+    test_data_dir = os.path.dirname(__file__)
+    with open(os.path.join(test_data_dir, 'test_data_1d.npz'), 'rb') as f:
+        buffer = io.BytesIO(f.read())
+        data = np.load(buffer)
+    x = data['x']
+    J = data['J']
+    Q = data['Q']
+    Sx0 = data['Sx']
+    # default
+    inputs0 = Input(shape=(x.shape[-1]))
+    scat0 = Scattering1D(J=J, Q=Q)(inputs0)
+    model0 = Model(inputs0, scat0)
+    model0.compile(optimizer='adam',
+                  loss='sparse_categorical_crossentropy',
+                  metrics=['accuracy'])
+    Sg0 = model0.predict(x)
+    assert np.allclose(Sg0, Sx0, atol=1e-07)
+    # adjust T
+    sigma_low_scale_factor = 2
+    T=2**(J-sigma_low_scale_factor)
+    inputs1 = Input(shape=(x.shape[-1]))
+    scat1 = Scattering1D(J=J, Q=Q, T=T)(inputs1)
+    model1 = Model(inputs1, scat1)
+    model1.compile(optimizer='adam',
+                  loss='sparse_categorical_crossentropy',
+                  metrics=['accuracy'])
+    Sg1 = model1.predict(x)
+    assert Sg1.shape == (Sg0.shape[0], Sg0.shape[1], Sg0.shape[2]*2**(sigma_low_scale_factor))
+
+    save_stdout = sys.stdout
+    result = io.StringIO()
+    sys.stdout = result
+    model1.summary()
```

Comment:
Isn't a subprocess better for capturing stdout?

Reply:
I dunno how to do that. I [used this stack exchange](https://stackoverflow.com/a/66310392) to figure this code above. I should credit it.

Reply:
On another stack exchange [post](https://stackoverflow.com/questions/5136611/capture-stdout-from-a-script) they posit multiple solutions to this problem, one of them being the subprocess. The method of using `StringIO` is also used. I think its fine.